utils.js

```diff
@@ -297,7 +262,33 @@ CTLEventUtils.populateURLParams = function(paramsArray) {
 };
 
 /**
- * This function takes in the location string from Bedewords and returns an
+ * This function returns the value from a paramsArray when given a key.
+ * If no value exists it returns null. If the value is a start or end
+ * date, it returns a date object.
+ *
+ * @param paramsArray = An object that contains key value representation
+ *                      of URL params. Call readURLParams to get these values.
+ *
+ * @param param = The key that you wish to fetch from the params array
+ *
+ * @return = The value associated with the key, if it exists
+ */
+CTLEventUtils.getURLParam = function(paramsArray, param) {
+    var value = null;
+    for (var i = 0; i < paramsArray.length; i++) {
+        if (paramsArray[i].key == param) {
+            value = paramsArray[i].value;
+            // if the param is supposed to be a date, instantiate a Date object
+            if (param == 'start' || param == 'end') {
+                value = new Date(value);
+            }
+        }
+    }
+    return value;
```

Comment:
I would also add `return value;` to after line 284, since you don't need to iterate through the rest of the array once the value is found.

Reply:
Got it, fixed.